### ISSUE TITLE
Implement remaining connection commands

### ISSUE THREAD

**Issue by mijicd** [zio/zio-redis#150]
Extracted from [this comment](https://github.com/zio/zio-redis/issues/68#issuecomment-691777714) authored by @regis-leray.

### Commands

- [ ] CLIENT CACHING YES|NO Instruct the server about tracking or not keys in the next request
- [ ] CLIENT ID Returns the client ID for the current connection
- [ ] CLIENT KILL [ip:port] [ID client-id] [TYPE normal|master|slave|pubsub] [USER username] [ADDR ip:port] [SKIPME yes/no] Kill the connection of a client
- [ ] CLIENT LIST [TYPE normal|master|replica|pubsub] Get the list of client connections
- [ ] CLIENT GETNAME Get the current connection name
- [ ] CLIENT GETREDIR Get tracking notifications redirection client ID if any
- [ ] CLIENT PAUSE timeout Stop processing commands from clients for some time
- [ ] CLIENT REPLY ON|OFF|SKIP Instruct the server whether to reply to commands
- [ ] CLIENT SETNAME connection-name Set the current connection name
- [ ] CLIENT TRACKING ON|OFF [REDIRECT client-id] [PREFIX prefix [PREFIX prefix ...]] [BCAST] [OPTIN] [OPTOUT] [NOLOOP] Enable or disable server assisted client side caching support
- [ ] CLIENT UNBLOCK client-id [TIMEOUT|ERROR] Unblock a client blocked in a blocking command from a different connection
- [ ] HELLO protover [AUTH username password] [SETNAME clientname] switch Redis protocol
- [ ] QUIT Close the connection

### Tips

- Define the API in `zio.redis.api`.
- Command-specific options should be defined in `zio.redis.options`.
- Additional inputs and outputs can be defined in `zio.redis.Input` and `zio.redis.Output`, respectively.
- Redis API evolves, therefore the commands listed above might be slightly changed. Double check [the official docs](https://redis.io/commands) before starting with implementation. 
- Use telnet to understand command protocol details.
- Write inputs and outputs tests in `zio.redis.InputSpec` and `zio.redis.OutputSpec`, respectively.
- Write integration tests by expanding the `zio.redis.ApiSpec`.
- If you feel that the pull request size is growing out of control, feel free to split it but make sure to link this issue in each of the related PRs.

**Comment by quelgar** [comment]
I think support for `CLIENT REPLY` should be a separate ticket, as it requires support in the executor to work. I think we'd need an API that puts the executor into "no reply/skip reply" mode, which would issue the `CLIENT REPLY` command and set some internal state to know not to expect replies.

**Comment by mijicd** [comment]
@quelgar Makes perfect sense. Let's pull that one out and tackle separately. It's something we can add post `v0.1.0` for sure :).

**Comment by mvelimir** [comment]
Hey @quelgar, can I take over this issue?

**Comment by quelgar** [comment]
> Hey @quelgar, can I take over this issue?

Go for it 👍
